workflow/Snakefile

```diff
@@ -40,11 +40,11 @@ BIOSAMPLE_ACTIVITES = {
 	biosample: activity for _, (biosample, activity) in 
 	BIOSAMPLE_DF[["biosample", "default_accessibility_feature"]].iterrows()
 }
+ABC_dirs = {biosample: os.path.join(RESULTS_DIR, biosample) for biosample in BIOSAMPLE_DF.biosamples}
```

Comment:
I wouldn't call this ABC_dirs since it stores e2g stuff as well. Probably `BIOSAMPLE_DIRS` is more fitting

Is the reason for this to shorten the code when doing os.path.joins? If it's just that, not sure if it's better since now you have to add the `lambda` fn when invoking it. Also it would be good to be consistent with the style instead of having some rules use ABC_dirs and others still have the old way

Reply:
I created this directory specifically to refer to the place where the directory where the EnhancersAllPutative and EnhancerList files live to make it compatible with running from an existing ABC directory if we want to implement that in the "apply" workflow in the future (this is something I did in the training workflow). In that case, the new features (SumCandidateEnhancers, etc.) will be created in a different location. I added the variable to this workflow to avoid having to copy the gen_new_features rule with different input paths for the two workflows.  So it's only meant to be used when referred when using those specific two files.

Reply:
makes sense, ty!

Can we rename to `ABC_BIOSAMPLES_DIR` to be more precise? (We also want to uppercase global variables).

And add a comment explaining the reason for this variable, e.g how training workflow works based on existing ABC directories?

Reply:
done!